R/rope.R

```diff
@@ -325,33 +337,77 @@ rope.stanfit <- rope.stanreg
 
 #' @rdname rope
 #' @export
-rope.brmsfit <- function(x, range = "default", ci = .89, ci_method = "HDI", effects = c("fixed", "random", "all"), component = c("conditional", "zi", "zero_inflated", "all"), parameters = NULL, verbose = TRUE, ...) {
+rope.brmsfit <- function(
+  x, 
+  range = "default", 
+  ci = .89, 
+  ci_method = "HDI", 
+  effects = c("fixed", "random", "all"), 
+  component = c("conditional", "zi", "zero_inflated", "all"), 
+  parameters = NULL, 
+  verbose = TRUE, 
+  ...
+) {
   effects <- match.arg(effects)
   component <- match.arg(component)
 
-  if (insight::is_multivariate(x)) {
-    stop("Multivariate response models are not yet supported.")
-  }
-
+  # check range argument
   if (all(range == "default")) {
     range <- rope_range(x)
+  # we expect a list with named vectors (length two) in the multivariate case.
+  # Names state the response variable.
+  } else if (insight::is_multivariate(x) && 
+    (!is.list(range) || length(range) != length(insight::find_response(x) ||
+        names(range) != insight::find_response(x)))) {
+    stop("With a multivariate model, `range` should be 'default' or a list of named numeric vectors with length 2.")
   } else if (!all(is.numeric(range)) || length(range) != 2) {
     stop("`range` should be 'default' or a vector of 2 numeric values (e.g., c(-0.1, 0.1)).")
   }
 
-  # check for possible collinearity that might bias ROPE
+  # check for possible collinearity that might bias ROPE and print a warning
   if (verbose) .check_multicollinearity(x, "rope")
 
-  rope_data <- rope(
-    insight::get_parameters(x, effects = effects, component = component, parameters = parameters),
-    range = range,
-    ci = ci,
-    ci_method = ci_method,
-    verbose = verbose,
-    ...
-  )
+  # calc rope
+  if (insight::is_multivariate(x)) {
+    dv <- insight::find_response(x)
+
+    # ROPE range / width differs between response varialbe. Thus ROPE is
+    # calculated for every variable on its own.
+    rope_data <- lapply(
+      dv,
+      function(dv) {
+        ret <- rope(
```

Comment:
I think this works, but I would not name the function argument `dv`, when the input object is also named `dv`.

Reply:
Thanks, fixed.